From 1b613476e5c6d7e888f1befaa1d3bb80e079be98 Mon Sep 17 00:00:00 2001
From: David Shea <reallylongword@gmail.com>
Date: Tue, 28 Feb 2023 15:18:26 -0500
Subject: [PATCH] Add CreatiVision support for screenscraper

---
 src/platform.cpp      | 7 +++++++
 src/screenscraper.cpp | 2 ++
 2 files changed, 9 insertions(+)

diff --git a/src/platform.cpp b/src/platform.cpp
index 70954ff..204d6c5 100644
--- a/src/platform.cpp
+++ b/src/platform.cpp
@@ -65,6 +65,7 @@ QStringList Platform::getPlatforms()
   platforms.append("channelf");
   platforms.append("coco");
   platforms.append("coleco");
+  platforms.append("crvision");
   platforms.append("daphne");
   platforms.append("dragon32");
   platforms.append("dreamcast");
@@ -191,6 +192,8 @@ QStringList Platform::getScrapers(QString platform)
     scrapers.append("screenscraper");
   } else if(platform == "coleco") {
     scrapers.append("screenscraper");
+  } else if(platform == "crvision") {
+    scrapers.append("screenscraper");
   } else if(platform == "daphne") {
     scrapers.append("screenscraper");
   } else if(platform == "dragon32") {
@@ -407,6 +410,8 @@ QString Platform::getFormats(QString platform, QString extensions, QString addEx
     formats.append("*.cas *.wav *.bas *.asc *.dmk *.jvc *.os9 *.dsk *.vdk *.rom *.ccc *.sna");
   } else if(platform == "coleco") {
     formats.append("*.bin *.col *.rom");
+  } else if(platform == "crvision") {
+    formats.append("*.bin *.rom");
   } else if(platform == "daphne") {
     formats.append("*.daphne");
   } else if(platform == "dragon32") {
@@ -597,6 +602,8 @@ QString Platform::getDefaultScraper(QString platform)
     scraper = "cache";
   } else if(platform == "coleco") {
     scraper = "cache";
+  } else if(platform == "crvision") {
+    scraper = "cache";
   } else if(platform == "daphne") {
     scraper = "cache";
   } else if(platform == "dragon32") {
diff --git a/src/screenscraper.cpp b/src/screenscraper.cpp
index 7bcdc3f..8d8c4c3 100644
--- a/src/screenscraper.cpp
+++ b/src/screenscraper.cpp
@@ -678,6 +678,8 @@ QString ScreenScraper::getPlatformId(const QString platform)
     return "144";
   } else if(platform == "coleco") {
     return "48";
+  } else if(platform == "crvision") {
+    return "241";
   } else if(platform == "daphne") {
     return "49";
   } else if(platform == "dragon32") {
-- 
2.39.1

