From 64883a04fd95c9cbab8579424d618f9d82f9abf8 Mon Sep 17 00:00:00 2001
From: David Shea <reallylongword@gmail.com>
Date: Thu, 26 Oct 2023 09:51:28 -0400
Subject: [PATCH 1/2] Remove hardcoded paths

---
 skyscraper.pro | 19 ++++++++++++-------
 src/main.cpp   |  2 +-
 2 files changed, 13 insertions(+), 8 deletions(-)

diff --git a/skyscraper.pro b/skyscraper.pro
index d016c10..56daf86 100755
--- a/skyscraper.pro
+++ b/skyscraper.pro
@@ -7,34 +7,39 @@ win32:CONFIG += console
 QT += core network xml
 QMAKE_CXXFLAGS += -std=c++11
 
-unix:target.path=/usr/local/bin
+!defined(BINDIR, var):BINDIR = "/usr/local/bin"
+unix:target.path=$$BINDIR
 unix:target.files=Skyscraper Skyscraper.app/Contents/MacOS/Skyscraper
 
-unix:supplementary.path=/usr/local/bin
+!defined(LIBEXECDIR, var):LIBEXECDIR = "/usr/local/bin"
+unix:supplementary.path=$$LIBEXECDIR/skyscraper
 unix:supplementary.files=\
   supplementary/scraperdata/check_screenscraper_json_to_idmap.py \
   supplementary/scraperdata/convert_platforms_json.py \
   supplementary/scraperdata/peas_and_idmap_verify.py
 
-unix:config.path=/usr/local/etc/skyscraper
+!defined(CONFDIR, var):CONFDIR = "/usr/local/etc/skyscraper"
+DEFINES+=CONFDIR=\\\"$$CONFDIR\\\"
+
+unix:config.path=$$CONFDIR
 unix:config.files=aliasMap.csv hints.xml mameMap.csv \
   mobygames_platforms.json peas.json platforms_idmap.csv \
   screenscraper_platforms.json tgdb_developers.json \
   tgdb_genres.json tgdb_platforms.json tgdb_publishers.json
 
-unix:examples.path=/usr/local/etc/skyscraper
+unix:examples.path=$$CONFDIR
 unix:examples.files=config.ini.example README.md artwork.xml \
   artwork.xml.example1 artwork.xml.example2 artwork.xml.example3 \
   artwork.xml.example4 docs/ARTWORK.md docs/CACHE.md
 
-unix:cacheexamples.path=/usr/local/etc/skyscraper/cache
+unix:cacheexamples.path=$$CONFDIR/cache
 unix:cacheexamples.files=cache/priorities.xml.example docs/CACHE.md
 
-unix:impexamples.path=/usr/local/etc/skyscraper/import
+unix:impexamples.path=$$CONFDIR/import
 unix:impexamples.files=docs/IMPORT.md import/definitions.dat.example1 \
   import/definitions.dat.example2
 
-unix:resexamples.path=/usr/local/etc/skyscraper/resources
+unix:resexamples.path=$$CONFDIR/resources
 unix:resexamples.files=resources/maskexample.png resources/frameexample.png \
   resources/boxfront.png resources/boxside.png resources/scanlines1.png \
   resources/scanlines2.png
diff --git a/src/main.cpp b/src/main.cpp
index b8331e0..cb3862a 100755
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -189,7 +189,7 @@ void setupUserConfig()
   QDir::setCurrent(skyDir.absolutePath());
 
   // copy configs
-  QString localEtcPath = QString("/usr/local/etc/skyscraper/");
+  QString localEtcPath = QString(CONFDIR) + QString("/");
 
   if (!QFileInfo::exists(localEtcPath)) {
     // RetroPie installation type
-- 
2.41.0

