From 01bc160d573108acf488e1b801dda5b5bd3d19e7 Mon Sep 17 00:00:00 2001
From: David Shea <reallylongword@gmail.com>
Date: Sun, 5 Mar 2023 23:30:48 -0500
Subject: [PATCH 4/4] Add Super Cassette Vision

---
 src/platform.cpp      | 7 +++++++
 src/screenscraper.cpp | 2 ++
 2 files changed, 9 insertions(+)

diff --git a/src/platform.cpp b/src/platform.cpp
index 051b42c..171b45a 100644
--- a/src/platform.cpp
+++ b/src/platform.cpp
@@ -115,6 +115,7 @@ QStringList Platform::getPlatforms()
   platforms.append("pv1000");
   platforms.append("saturn");
   platforms.append("scummvm");
+  platforms.append("scv");
   platforms.append("sega32x");
   platforms.append("segacd");
   platforms.append("sg-1000");
@@ -302,6 +303,8 @@ QStringList Platform::getScrapers(QString platform)
   } else if(platform == "scummvm") {
     scrapers.append("screenscraper");
     scrapers.append("thegamesdb");
+  } else if(platform == "scv") {
+    scrapers.append("screenscraper");
   } else if(platform == "sega32x") {
     scrapers.append("screenscraper");
   } else if(platform == "segacd") {
@@ -516,6 +519,8 @@ QString Platform::getFormats(QString platform, QString extensions, QString addEx
     formats.append("*.cue *.iso *.mdf *.chd *.m3u");
   } else if(platform == "scummvm") {
     formats.append("*.svm *.scummvm");
+  } else if(platform == "scv") {
+    formats.append("*.bin");
   } else if(platform == "sega32x") {
     formats.append("*.32x *.bin *.md *.smd");
   } else if(platform == "segacd") {
@@ -712,6 +717,8 @@ QString Platform::getDefaultScraper(QString platform)
     scraper = "cache";
   } else if(platform == "scummvm") {
     scraper = "cache";
+  } else if(platform == "scv") {
+    scraper = "cache";
   } else if(platform == "sega32x") {
     scraper = "cache";
   } else if(platform == "segacd") {
diff --git a/src/screenscraper.cpp b/src/screenscraper.cpp
index 0c433bf..477748b 100644
--- a/src/screenscraper.cpp
+++ b/src/screenscraper.cpp
@@ -778,6 +778,8 @@ QString ScreenScraper::getPlatformId(const QString platform)
     return "22";
   } else if(platform == "scummvm") {
     return "123";
+  } else if(platform == "scv") {
+    return "67";
   } else if(platform == "sega32x") {
     return "19";
   } else if(platform == "segacd") {
-- 
2.39.1

