From d70f9e9a5030ce8b0c52eae0a2c71fb27d4975b9 Mon Sep 17 00:00:00 2001
From: David Shea <reallylongword@gmail.com>
Date: Wed, 8 Mar 2023 19:16:27 -0500
Subject: [PATCH 5/5] Add GX4000

---
 src/platform.cpp      | 7 +++++++
 src/screenscraper.cpp | 2 ++
 2 files changed, 9 insertions(+)

diff --git a/src/platform.cpp b/src/platform.cpp
index 171b45a..510c35c 100644
--- a/src/platform.cpp
+++ b/src/platform.cpp
@@ -80,6 +80,7 @@ QStringList Platform::getPlatforms()
   platforms.append("gbc");
   platforms.append("gc");
   platforms.append("genesis");
+  platforms.append("gx4000");
   platforms.append("intellivision");
   platforms.append("mame-advmame");
   platforms.append("mame-libretro");
@@ -226,6 +227,8 @@ QStringList Platform::getScrapers(QString platform)
     scrapers.append("screenscraper");
   } else if(platform == "genesis") {
     scrapers.append("screenscraper");
+  } else if(platform == "gx4000") {
+    scrapers.append("screenscraper");
   } else if(platform == "intellivision") {
     scrapers.append("screenscraper");
   } else if(platform == "mame-advmame") {
@@ -449,6 +452,8 @@ QString Platform::getFormats(QString platform, QString extensions, QString addEx
     formats.append("*.iso *.cso *.gcz *.gcm *.m3u *.rvz");
   } else if(platform == "genesis") {
     formats.append("*.smd *.bin *.gen *.md *.sg");
+  } else if(platform == "gx4000") {
+    formats.append("*.cpr *.dsk *.m3u");
   } else if(platform == "intellivision") {
     formats.append("*.int *.bin");
   } else if(platform == "mame-advmame") {
@@ -647,6 +652,8 @@ QString Platform::getDefaultScraper(QString platform)
     scraper = "cache";
   } else if(platform == "genesis") {
     scraper = "cache";
+  } else if(platform == "gx4000") {
+    scraper = "cache";
   } else if(platform == "intellivision") {
     scraper = "cache";
   } else if(platform == "megacd") {
diff --git a/src/screenscraper.cpp b/src/screenscraper.cpp
index 477748b..eef51ad 100644
--- a/src/screenscraper.cpp
+++ b/src/screenscraper.cpp
@@ -708,6 +708,8 @@ QString ScreenScraper::getPlatformId(const QString platform)
     return "13";
   } else if(platform == "genesis") {
     return "1";
+  } else if(platform == "gx4000") {
+    return "87";
   } else if(platform == "intellivision") {
     return "115";
   } else if(platform == "mame-advmame") {
-- 
2.39.1

